bridge/setu/broadcaster/broadcaster.go

```diff
@@ -52,41 +59,35 @@ func (tb *TxBroadcaster) BroadcastToHeimdall(msg sdk.Msg) error {
 	txEncoder := helper.GetTxEncoder(tb.cliCtx.Codec)
 	// chain id
 	chainID := helper.GetGenesisDoc().ChainID
-	// current address
-	address := hmTypes.BytesToHeimdallAddress(helper.GetAddress())
+
 	// fetch from APIs
-	var account authTypes.Account
-	response, err := util.FetchFromAPI(tb.cliCtx, util.GetHeimdallServerEndpoint(fmt.Sprintf(util.AccountDetailsURL, address)))
+	account, err := util.GetAccount(tb.cliCtx)
```

Comment:
No need to fetch account everytime. You can fetch in case of error for fresh sequence.

Reply:
Done